lib/api/dsl/index.js

```diff
@@ -1,117 +1,67 @@
+  Filters = require('./filters'),
   NotFoundError = require('kuzzle-common-objects').Errors.notFoundError,
+  BadRequestError = require('kuzzle-common-objects').Errors.badRequestError,
   async = require('async'),
   _ = require('lodash'),
-  q = require('q');
-
-
-module.exports = function Dsl (kuzzle) {
-  /**
-   * A tree where we have an entry by collection, an entry by tag and
-   * an entry by filter, with the rooms list and the corresponding operator arguments
-   *
-   * @example
-   * Example for chat-room-kuzzle (see above)
-   *  filtersTree = {
-   *    index : { // -> index name
-   *      collection : { // -> collection name
-   *        rooms: [] // -> global room to test each time (for a a subscribe on a whole collection, or if 'not exists' filter (see issue #1 on github)
-   *        fields: {
-   *          subject : { // -> attribute where a filter exists
-   *            termSubjectKuzzle : {
-   *              rooms: [ 'f45de4d8ef4f3ze4ffzer85d4fgkzm41'], // -> room id that match this filter
-   *              args: {operator, not, field, value}
-   *            }
-   *          }
-   *        }
-   *      }
-   *    }
-   *  }
-   */
-  this.filtersTree = {};
+  q = require('q'),
+  stringify = require('json-stable-stringify'),
+  crypto = require('crypto');
 
-  this.methods = require('./methods');
-  this.methods.dsl = this;
-  this.kuzzle = kuzzle;
+module.exports = function Dsl () {
+  this.filters = new Filters();
 
   /**
-   * Creates new entries in the filtersTree with the provided filters
+   * Create an unique filter ID from an user filter.
+   * The calculation is predictable, meaning the resulting
+   * filter ID will always be the same with identical filters,
+   * independently of the terms order.
    *
-   * @param {string} roomId
-   * @param {string} index
-   * @param {string} collection
-   * @param {Object} filters
-   * @returns {promise} the generated method
+   * @param index
+   * @param collection
+   * @param filters
+   * @results {string} filter unique ID
    */
-  this.addSubscription = function (roomId, index, collection, filters) {
-    var
-      filterName,
-      privateFilterName;
-    
-    if (filters === undefined) {
-      return q.reject(new BadRequestError('Filters parameter can\'t be undefined'));
-    }
+  this.createFilterId = function (index, collection, filters) {
+    var idObject = {index, collection};
 
-    filterName = Object.keys(filters)[0];
+    idObject.filters = filters || {};
 
-    if (filterName === undefined) {
-      return q.reject(new BadRequestError('Undefined filters'));
-    }
-
-    privateFilterName = _.camelCase(filterName);
-
-    if (!methods[privateFilterName]) {
-      return q.reject(new NotFoundError('Unknown filter with name '+ privateFilterName));
-    }
-    
-    return this.methods[privateFilterName](roomId, index, collection, filters[filterName]);
+    return crypto.createHash('md5').update(stringify(idObject)).digest('hex');
```

Comment:
Same answer than the one I gave to @benoitvidis : it's redundant with `json-stable-stringify` :smiley: